### PR TITLE
Support nested clip volumes

### PR DESCRIPTION
**Pull request by pmconne**
Instead of each ClipVolume having its own texture, each just has a Float32Array (or Uint8Array, on platforms that lack support for floating point textures) containing the encoded clip planes.
A Target's BranchUniforms has a ClipStack that has a texture. As ClipVolumes are pushed onto the stack, their encoded data is appended to the texture. So the texture ends up containing the intersection of any number of UnionOfConvexClipPlaneSets. The texture grows as needed but never shrinks.
`ViewFlags.clipVolume` refers specifically to the **view** clip; other clips (from ModelClipGroups, schedule scripts, reality tiles, etc) always apply.

Confirmed that clips from ModelClipGroups, schedule scripts, and reality tile overlap logic all nest correctly now.

Incidentally promoted ModelClipGroups and RenderClipVolume to `public`.
Added `insideColor` and `outsideColor` properties to ClipStyle, replacing the non-persistent Viewport.setInside/OutsideClipColor methods.

[Backlog item](https://bentleycs.visualstudio.com/iModelTechnologies/_workitems/edit/577198).

### CODE REVIEW

File: core/frontend/src/render/webgl/ClipStack.ts [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+  const p = scratchRangeCorners;
+  p[0].setFromPoint3d(r.low);
+  p[1].set(r.high.x, r.low.y, r.low.z);
+  p[2].set(r.low.x, r.high.y, r.low.z);
+  p[3].set(r.high.x, r.high.y, r.low.z);
+  p[4].set(r.low.x, r.low.y, r.high.z);
+  p[5].set(r.high.x, r.low.y, r.high.z);
+  p[6].set(r.low.x, r.high.y, r.high.z);
+  p[7].setFromPoint3d(r.high);
+  return p;
```
Comment (RBBentley) [inline core/frontend/src/render/webgl/ClipStack.ts:46]: Couldn't you just return r.corners(scratchRangeCorners) here?

Comment (RBBentley) [inline core/frontend/src/render/webgl/ClipStack.ts:46]: Couldn't you just return r.corners(scratchRangeCorners) here?

Comment (pmconne) [reply core/frontend/src/render/webgl/ClipStack.ts:46]: Didn't know that was a thing.

Comment (pmconne) [reply core/frontend/src/render/webgl/ClipStack.ts]: Done.

**Comment by markschlosseratbentley** [comment]
@pmconne Was this tested anywhere besides Windows?

**Comment by pmconne** [comment]
> @pmconne Was this tested anywhere besides Windows?

No.

**Comment by markschlosseratbentley** [comment]
> > @pmconne Was this tested anywhere besides Windows?
> 
> No.

@pmconne dta iOS is actually currently broken using master - there's a pending PR wip to fix it, so I'll make a note to test this soon (shouldn't hold up this PR imo).

**Review by RBBentley — APPROVED** [review]
Looks good... I validated the incremental tile display for reality tiles continues to function.

**Review by markschlosseratbentley — APPROVED** [review]
Code looks good; I also tested and it works for me on desktop Windows.  I'll try to test on iOS in the near future.

**Comment by pmconne** [comment]
@Mergifyio backport release/2.15.x

**Pull request merged**